cranelift-codegen/src/context.rs

```diff
@@ -132,33 +132,37 @@ impl Context {
         self.verify_if(isa)?;
         debug!("Compiling:\n{}", self.func.display(isa));
 
+        let opt_level = isa.flags().opt_level();
+
         self.compute_cfg();
-        if isa.flags().opt_level() != OptLevel::Fastest {
+        if opt_level != OptLevel::Fastest {
             self.preopt(isa)?;
         }
         if isa.flags().enable_nan_canonicalization() {
             self.canonicalize_nans(isa)?;
         }
         self.legalize(isa)?;
-        if isa.flags().opt_level() != OptLevel::Fastest {
+        if opt_level != OptLevel::Fastest {
             self.postopt(isa)?;
         }
-        if isa.flags().opt_level() == OptLevel::Best {
+        if opt_level == OptLevel::NearlyBest || opt_level == OptLevel::Best {
             self.compute_domtree();
             self.compute_loop_analysis();
             self.licm(isa)?;
```

Comment:
I believe I read somewhere LICM caused a wasm bench with clif to be slower. Also it being broken (#732) is the reason I don't enable opt_level=best in cg_clif.

Reply:
Yeah, it's time to ping Dan...